cpg-core/src/test/kotlin/de/fraunhofer/aisec/cpg/enhancements/templates/ClassTemplateTest.kt

```diff
@@ -139,29 +140,28 @@ internal class ClassTemplateTest : BaseTest() {
     fun testClassTemplateStructure() {
         val result =
             analyze(listOf(Path.of(topLevel.toString(), "pair.cpp").toFile()), topLevel, true)
-        val classTemplateDeclarations = flattenListIsInstance<ClassTemplateDeclaration>(result)
+        val classTemplateDeclarations = result.allChildren<ClassTemplateDeclaration>()
         val template =
             findByUniqueName(
                 classTemplateDeclarations,
                 "template<class Type1, class Type2> class Pair"
             )
-        val pair = findByUniqueName(flattenListIsInstance<RecordDeclaration>(result), "Pair")
-        val type1 =
-            findByUniqueName(flattenListIsInstance<TypeParamDeclaration>(result), "class Type1")
-        val type2 =
-            findByUniqueName(flattenListIsInstance<TypeParamDeclaration>(result), "class Type2")
-        val first = findByUniqueName(flattenListIsInstance<FieldDeclaration>(result), "first")
-        val second = findByUniqueName(flattenListIsInstance<FieldDeclaration>(result), "second")
+        val pair = findByUniqueName(result.records, "Pair")
+        val type1 = findByUniqueName(result.allChildren<TypeParamDeclaration>(), "class Type1")
+        val type2 = findByUniqueName(result.allChildren<TypeParamDeclaration>(), "class Type2")
+        val first = findByUniqueName(result.allChildren<FieldDeclaration>(), "first")
+        val second = findByUniqueName(result.allChildren<FieldDeclaration>(), "second")
         val receiver = pair.byNameOrNull<MethodDeclaration>("Pair")?.receiver
         assertNotNull(receiver)
 
         val pairConstructorDeclaration =
-            findByUniqueName(flattenListIsInstance<ConstructorDeclaration>(result), "Pair")
+            findByUniqueName(result.allChildren<ConstructorDeclaration>(), "Pair")
         val constructExpression =
-            findByUniquePredicate(flattenListIsInstance(result)) { c: ConstructExpression ->
+            findByUniquePredicate(result.allChildren<ConstructExpression>()) {
+                c: ConstructExpression ->
                 c.code == "()"
             }
-        val point1 = findByUniqueName(flattenListIsInstance<VariableDeclaration>(result), "point1")
+        val point1 = findByUniqueName(result.allChildren<VariableDeclaration>(), "point1")
```

Comment:
```suggestion
        val point1 = result.variables["point1"]!!
```

Reply:
I will leave changing the findByUnique for a later PR.